src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -98,10 +103,17 @@ private void VisitArrayType(IArrayTypeSymbol symbol, bool? isNullable)
 
                 AddArrayRank(arrayType);
 
+                // If we want to display `!`, then we surely also want to display `?`
+                Debug.Assert(format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeNullableTypeModifier)
```

Comment:
> Debug.Assert(format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeNullableTypeModifier) [](start = 15, length = 125)

Same comment as above. Consider moving this assert into SymbolDisplayFormat constructor instead of duplicating it in the visitor multiple times. When this assert fails we will be far from the place that created SymbolDisplayFormat with inconsistent options.